src/asynqp/connection.py

```diff
@@ -60,9 +60,10 @@ def close(self):
 
         This method is a :ref:`coroutine <coroutine>`.
         """
-        self.sender.send_Close(0, 'Connection closed by application', 0, 0)
-        yield from self.synchroniser.await(spec.ConnectionCloseOK)
-        self.closed.set_result(True)
+        if not self.closed.done():
+            self.sender.send_Close(0, 'Connection closed by application', 0, 0)
+            yield from self.synchroniser.await(spec.ConnectionCloseOK)
+            self.closed.set_result(True)
```

Comment:
I wonder if, instead, this should throw an exception if the connection is already closed.